erpnext/stock/doctype/stock_reconciliation/stock_reconciliation.py

```diff
@@ -465,20 +465,53 @@ def get_items(warehouse, posting_date, posting_time, company):
 	""", (lft, rgt, company))
 
 	res = []
-	for d in set(items):
+	for d in items:
 		stock_bal = get_stock_balance(d[0], d[2], posting_date, posting_time,
 			with_valuation_rate=True)
 
 		if frappe.db.get_value("Item", d[0], "disabled") == 0:
-			res.append({
-				"item_code": d[0],
-				"warehouse": d[2],
-				"qty": stock_bal[0],
-				"item_name": d[1],
-				"valuation_rate": stock_bal[1],
-				"current_qty": stock_bal[0],
-				"current_valuation_rate": stock_bal[1]
-			})
+			package_tags = frappe.get_all("Stock Ledger Entry",
+				filters={"item_code": d[0], "warehouse": d[2], "package_tag": ["is", "set"]},
+				fields=["distinct(package_tag)", "actual_qty", "valuation_rate"])
+
+			batch_nos = frappe.get_all("Stock Ledger Entry",
+				filters={"item_code": d[0], "warehouse": d[2], "batch_no": ["is", "set"]},
+				fields=["distinct(batch_no)", "actual_qty", "valuation_rate"])
+
+			if package_tags:
+				for package_tag in package_tags:
+					res.append({
+						"item_code": d[0],
+						"warehouse": d[2],
+						"package_tag": package_tag.package_tag,
+						"qty": package_tag.actual_qty,
+						"item_name": d[1],
+						"valuation_rate": package_tag.valuation_rate,
+						"current_qty": stock_bal[0],
+						"current_valuation_rate": stock_bal[1]
+					})
+			if batch_nos:
```

Comment:
```suggestion
			elif batch_nos:
```

Reply:
kept this if purposely as there might be case where item has Batches and Package Tags both.